### ISSUE TITLE
Models don't import, just a plane and an empty

### ISSUE THREAD

**Issue by ReeceTaylor512** [TobyLobster/ImportLDraw#73]
Been trying to get this addon to work but whenever I do it never imports properly. I just get a plane and an empty

![image](https://user-images.githubusercontent.com/73254003/204402876-c570b92b-bb7e-499a-a61d-18be97152343.png) 

I've added the filepath
![image](https://user-images.githubusercontent.com/73254003/204403031-877550ec-b4e3-4658-af78-cf22f1be3316.png)

And here is what that filepath looks like
![image](https://user-images.githubusercontent.com/73254003/204403145-2802cca5-f19a-4fe8-82ad-b739f2de2793.png)

**Comment by meteorshowers** [comment]
the same problem

**Comment by Xolverr** [comment]
Same problem here

**Comment by TobyLobster** [comment]
Ok thanks @ReeceTaylor512, let me know when you have some time.

**Comment by ReeceTaylor512** [comment]
Hey, I'm available now. Where's the best place to discuss this? I can do it here or somewhere else like discord if you prefer

**Comment by TobyLobster** [comment]
Hi, thanks for getting back. I'm happy to keep going on here. 

I attach the latest version (as yet unreleased) without the huge background.exr file that's normally part of the release but is too large to attach here. The lack of this file means it may not give a pretty renders as a result (colours may be wrong) but should otherwise work OK. 

A couple of things that would be helpful if you can. Firstly if you could try installing this latest version and confirm if it works or not, that would be good to check. (I suspect the problem is still present though). 

Secondly, I don't know if you are familiar with the Command Prompt in Windows? If Blender is started from the command prompt then you can see some debug output when my plugin executes which would be helpful to me in narrowing down where the problem lies. What you need to do is: 1. Open the Command Prompt. 2. Type in the full path to the Blender executable. 3. Press return. Blender should launch. 4. Try using this LDraw plugin. 5. Look back at the Command Prompt for the output.

This might take a few iterations back and forth between us to narrow this down, but if you can, thanks for your help.

[importldraw1.1.18_for_blender_281.zip](https://github.com/TobyLobster/ImportLDraw/files/11727801/importldraw1.1.18_for_blender_281.zip)

**Comment by Vap0r1ze** [comment]
I get this issue too on Windows 10. Some helpful info: on windows you don't need to run blender in cmd to see debug logs, you can toggle the cmd shell inside Blender with **Window > Toggle System Console**.

Here's a log of me attempting to import the attached LDraw file ([canetest.zip](https://github.com/TobyLobster/ImportLDraw/files/12038584/canetest.zip)):
<pre>
Read prefs: C:\Users\super\AppData\Roaming\Blender Foundation\Blender\2.93\config\userpref.blend
Registered Animation Nodes
addon_utils.disable: io_scene_importldraw not disabled
Modules Installed (io_scene_importldraw) from 'C:\\Users\\super\\Downloads\\importldraw1.1.18_for_blender_281.zip' into 'C:\\Users\\super\\AppData\\Roaming\\Blender Foundation\\Blender\\2.93\\scripts\\addons'
06:52:18.80 [importldraw] The LDraw Parts Library path to be used is: /Applications/Lego/Bricksmith/ldraw
06:52:18.80 [importldraw] Use LSynth Parts requested
06:52:18.80 [importldraw] Standard-res primitives selected
06:52:18.80 [importldraw] Loading stud files
06:52:18.81 [importldraw] Loading files
06:52:18.82 [importldraw] WARNING: Missing file 41239.dat
06:52:18.82 [importldraw] WARNING: Missing file 32524.dat
06:52:18.82 [importldraw] WARNING: Missing file 6558.dat
06:52:18.83 [importldraw] WARNING: Missing file 40490.dat
06:52:18.83 [importldraw] WARNING: Missing file 32278.dat
06:52:18.83 [importldraw] WARNING: Missing file 14720.dat
06:52:18.83 [importldraw] WARNING: Missing file 32054.dat
06:52:18.83 [importldraw] WARNING: Missing file 32062.dat
06:52:18.83 [importldraw] WARNING: Missing file 6538b.dat
06:52:18.83 [importldraw] WARNING: Missing file 43093.dat
06:52:18.84 [importldraw] WARNING: Missing file 62462.dat
06:52:18.84 [importldraw] WARNING: Missing file 89678.dat
06:52:18.84 [importldraw] WARNING: Missing file 6141.dat
06:52:18.84 [importldraw] WARNING: Missing file 55013.dat
06:52:18.84 [importldraw] WARNING: Missing file 3713.dat
06:52:18.84 [importldraw] WARNING: Missing file 3062b.dat
06:52:18.85 [importldraw] WARNING: Missing file 3004.dat
06:52:18.85 [importldraw] WARNING: Missing file 3622.dat
06:52:18.85 [importldraw] WARNING: Missing file 3010.dat
06:52:18.85 [importldraw] WARNING: Missing file 3005.dat
06:52:18.85 [importldraw] WARNING: Missing file 3710.dat
06:52:18.85 [importldraw] WARNING: Missing file 3623.dat
06:52:18.85 [importldraw] Creating NodeGroups
06:52:18.85 [importldraw] createBlenderDistanceToCenterNodeGroup #create
06:52:18.86 [importldraw] createBlenderVectorElementPowerNodeGroup #create
06:52:18.86 [importldraw] createBlenderConvertToNormalsNodeGroup #create
06:52:18.86 [importldraw] createBlenderConcaveWallsNodeGroup #create
06:52:18.87 [importldraw] createBlenderSlopeTextureNodeGroup #create
06:52:18.87 [importldraw] createBlenderFresnelNodeGroup #create
06:52:18.87 [importldraw] createBlenderReflectionNodeGroup #create
06:52:18.87 [importldraw] createBlenderDielectricNodeGroup #create
06:52:18.87 [importldraw] createBlenderLegoStandardNodeGroup #create
06:52:18.87 [importldraw] createBlenderLegoTransparentNodeGroup #create
06:52:18.87 [importldraw] createBlenderLegoTransparentFluorescentNodeGroup #create
06:52:18.87 [importldraw] createBlenderLegoRubberNodeGroup #create
06:52:18.87 [importldraw] createBlenderLegoRubberTranslucentNodeGroup #create
06:52:18.88 [importldraw] createBlenderLegoEmissionNodeGroup #create
06:52:18.88 [importldraw] createBlenderLegoChromeNodeGroup #create
06:52:18.88 [importldraw] createBlenderLegoPearlescentNodeGroup #create
06:52:18.88 [importldraw] createBlenderLegoMetalNodeGroup #create
06:52:18.88 [importldraw] createBlenderLegoGlitterNodeGroup #create
06:52:18.88 [importldraw] createBlenderLegoSpeckleNodeGroup #create
06:52:18.88 [importldraw] createBlenderLegoMilkyWhiteNodeGroup #create
06:52:18.88 [importldraw] Creating Blender objects
06:52:18.92 [importldraw] Number of vertices: 0
06:52:18.92 [importldraw] Number of convex hull vertices: 0
06:52:18.92 [importldraw] Adding 659 objects to scene
06:52:19.05 [importldraw] Load Done
</pre>

> `The LDraw Parts Library path to be used is: /Applications/Lego/Bricksmith/ldraw`
I'm not familiar with how this works, but it looks like the issue might be that it tries to point to the wrong directory for this library on Windows. I'll do some snooping for the Windows path and edit the respective script to see if that fixes my issue.

**Comment by Vap0r1ze** [comment]
After a little snooping it looks like the issue I'm seeing is that when you upload releases you are uploading your personal `ImportLDrawPreferences.ini` file along with the add-on bundle. I deleted this file and let it regenerate, and now I've moxed onto a different issue (progress!)

**Comment by ReeceTaylor512** [comment]
Yeah sorry it's been a while for me life has been busier than expected. I did manage to test the importer with a model from Stud.io and the same model from LDD and what's strange is that it imported from Stud.io but not LDD

**Comment by JHubi1** [comment]
Hi! Just wanted to ask if there is any solution. I'm using Studio 2.0 and the following error is occurring when I try to import a model. I even had to make a screen record to capture it (it disappeared that fast).
The log says:
```
[...]
00:45:15.71 [importldraw] Loading files
00:45:15.71 [importldraw] WARNING: Missing file 3815c01.dat
Missing file 3815c01.dat
00:45:15.71 [importldraw] WARNING: Missing file bl_973pb2342c01.dat
Missing file bl_973pb2342c01.dat
00:45:15.71 [importldraw] WARNING: Missing file 92081.dat
Missing file 92081.dat
00:45:15.71 [importldraw] WARNING: Missing file 3626cp88.dat
Missing file 3626cp88.dat
[...]
```

Video:
https://github.com/TobyLobster/ImportLDraw/assets/61345690/0874a92a-402a-41fe-8945-d1d2455cabaa

**Comment by ReeceTaylor512** [comment]
Working fine for me with studio models. Having an issue with LDD parts though.

@JHubi1 are you sure you added the right path to the LDRAW library?

**Comment by JHubi1** [comment]
No, sorry, I wasn't into it at that moment, after a bit of trying around, I was able to make it run. The path was set wrong, so easy it can be 😅

**Comment by Vap0r1ze** [comment]
> After a little snooping it looks like the issue I'm seeing is that when you upload releases you are uploading your personal `ImportLDrawPreferences.ini` file along with the add-on bundle.

Would like to bring up what I wrote again, it seems to have been forgotten

**Comment by SaffronStreams** [comment]
I had the same problem as described here, but after reading some comments and rereading the install instructions, I realized I had misunderstood the "Setting the LDraw Parts Library directory"instructions.

Where it says "Type the full filepath to the 'ldraw' directory you unzipped to." I thought it was a way of saying "navigate to where you have the library and import one of those files", but it's actually the path you need to set under import options, you're not importing anything at that time, you're just setting the path. Once you set it there, it does seem to remember it, so that's nice.

<img width="611" alt="image" src="https://github.com/TobyLobster/ImportLDraw/assets/50481103/d91fec51-252b-42f4-adf9-aaa657e6bfb8">

With that path set, you can now navigate to where you put your ldr file you wanted to import, and import it, making sure your ldraw filepath is set to the library. I did still have missing hair, but everything else seemed to have import the way I wanted it to 

![image](https://github.com/TobyLobster/ImportLDraw/assets/50481103/77f19be0-8a1d-4ec8-b88f-4e18939b470a)

**Comment by cjhendrix** [comment]
> > After a little snooping it looks like the issue I'm seeing is that when you upload releases you are uploading your personal `ImportLDrawPreferences.ini` file along with the add-on bundle.
> 
> Would like to bring up what I wrote again, it seems to have been forgotten

Thanks @Vap0r1ze Your fix worked for me in Blender 3.6 with v1.18 of the addon.

**Comment by TobyLobster** [comment]
I have just released a new version fro Blender 4 that should fix this issue.
